chain/chain/src/chain.rs

```diff
@@ -1843,6 +1844,35 @@ impl Chain {
         Ok(())
     }
 
+    pub fn build_state_for_split_shards(
+        &mut self,
+        sync_hash: &CryptoHash,
+        shard_id: ShardId,
+    ) -> Result<(), Error> {
+        let (epoch_id, next_epoch_id) = {
+            let block_header = self.get_block_header(sync_hash)?;
+            (block_header.epoch_id().clone(), block_header.next_epoch_id().clone())
+        };
+        let state_root = {
+            let block = self.get_block(&sync_hash)?;
+            block
+                .chunks()
+                .get(shard_id as usize)
+                .ok_or(ErrorKind::InvalidShardId(shard_id))?
+                .prev_state_root()
+        };
```

Comment:
What happens when the first block of the epoch (sync hash) does not contain a new chunk for some shard? According to what is done here, we will ignore the state transition happened in the last chunk for that shard, which is not acceptable.

Reply:
That's a very good point. I forgot about that. I will get state root from chunk extra